resim/metrics/python/emissions.py

```diff
@@ -128,3 +197,492 @@ def emit(
+        non_list_fields = [k for k, v in data.items() if not isinstance(v, list)]
+        if non_list_fields:
+            non_list_str = ", ".join(f"'{f}'" for f in non_list_fields)
+            raise ReSimValidationError(
+                f"all data values must be lists in emit_series, but these fields are not: {non_list_str}",
+                topic=topic_name,
+            )
+
+        # Determine the number of points
+        if timestamps is not None:
+            # Validate lengths match when timestamps are provided
+            num_points = len(timestamps)
+
+            mismatched = [(k, len(v)) for k, v in data.items() if len(v) != num_points]
+            if mismatched:
+                length_info = ", ".join(f"'{k}': {length}" for k, length in mismatched)
+                raise ReSimValidationError(
+                    f"all series must be the same length as timestamps (length {num_points}), "
+                    f"but these fields have different lengths: {length_info}",
+                    topic=topic_name,
+                )
+        else:
+            # No timestamps provided - all series should be the same length
+            lengths = {k: len(v) for k, v in data.items()}
+            unique_lengths = set(lengths.values())
+            if len(unique_lengths) != 1:
+                length_info = ", ".join(
+                    f"'{k}': {length}" for k, length in lengths.items()
+                )
+                raise ReSimValidationError(
+                    f"all series must be the same length in emit_series, but got different lengths: {length_info}",
+                    topic=topic_name,
+                )
+            num_points = unique_lengths.pop()
+
+        # Validate each point in the series
+        for i in range(num_points):
+            point_data = {k: v[i] for k, v in data.items()}
+            try:
+                self._validate_data(topic_name, point_data)
+            except ReSimValidationError as e:
+                # Re-raise with series index context
+                raise ReSimValidationError(
+                    str(e),
+                    topic=topic_name,
+                    series_index=i,
+                ) from e
+
+    def _validate_value(
+        self, value: Any, expected_type_str: str, field_name: str, topic_name: str
+    ) -> None:
+        """
+        Validate a single value against an expected type.
+
+        Args:
+            value: The value to validate.
+            expected_type_str: The expected type as a string (e.g., "int", "string[]").
+            field_name: The name of the field being validated (for error messages).
+            topic_name: The name of the topic (for error messages).
+
+        Raises:
+            ReSimValidationError: If the value does not match the expected type.
+        """
+        if expected_type_str not in self.TYPE_MAPPING:
+            raise ReSimValidationError(
+                f"unknown type '{expected_type_str}'",
```

Comment:
```suggestion
                f"unknown type '{expected_type_str}' implied by the config",
```
In my mind this helps because it tells you that it's not something that you emitted thats wrong, but the config itself. Not sure if I'm missing something though.